readthedocs/api/v3/views.py

```diff
@@ -297,6 +298,33 @@ def get_serializer_class(self):
             return VersionSerializer
         return VersionUpdateSerializer
 
+    def update(self, request, *args, **kwargs):
+        """
+        Run extra steps after updating a version.
+
+        - When a version is deactivated, we need to clean up its
+          files from storage, and search index.
+        - When a version is activated, we need to trigger a build.
+        - We also need to purge the cache from the CDN,
+          since the version could have been activated/deactivated,
+          or its privacy level could have changed.
+        """
+        # Get the current value before updating.
+        version = self.get_object()
+        was_active = version.active
+        result = super().update(request, *args, **kwargs)
+        # Get the updated version.
+        version = self.get_object()
+        # If the version is deactivated, we need to clean up the files.
+        if was_active and not version.active:
+            version.clean_resources()
+        # If the version is activated, we need to trigger a build.
+        if not was_active and version.active:
+            trigger_build(project=version.project, version=version)
```

Comment:
Is it possible to share this logic with the Form so we don't have to maintain it in both places?

Reply:
I'm not sure where the functionality comes from, but activating a new version also triggers a build.

Reply:
@benjaoming that is done from the Version form.

@humitos I was able to unify the call from the form and the view, now this is just a call to a method from the version.